Does not handle emulated x64 Python gracefully on Windows ARM

Native support for CPython on Windows ARM was introduced in Python 3.11. For all versions before that, Windows ARM will perform x64 emulation.

```
[25](https://github.com/winstxnhdw/gxhash/actions/runs/24291328511/job/70929112833#step:6:226)
DEBUG 💥 maturin failed
DEBUG   Caused by: Failed to build a native library through cargo
DEBUG   Caused by: Cargo build finished with "exit code: 101": `"cargo" "rustc" "--profile" "release" "--features" "pyo3/extension-module" "--target" "x86_64-pc-windows-msvc" "--message-format" "json-render-diagnostics" "--locked" "--manifest-path" "C:\\a\\_temp\\setup-uv-cache\\sdists-v9\\path\\938a27ddb9b6d1d7\\LWa855XvUxlnvSQj49vzG\\src\\Cargo.toml" "--lib" "--crate-type" "cdylib" "--" "-C" "strip=symbols"`
DEBUG Error: command ['maturin', 'pep517', 'build-wheel', '-i', 'C:\\a\\_temp\\setup-uv-cache\\builds-v0\\.tmpv7xmj3\\Scripts\\python.exe', '--compatibility', 'off'] returned non-zero exit status 1
  × Failed to build `gxhash @
  │ file:///C:/a/gxhash/gxhash/dist/gxhash-0.6.0.tar.gz`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `maturin.build_wheel` failed (exit code: 1)

      [stdout]
      cargo 1.95.0-nightly (85eff7c80 2026-01-15)
      Rust not found, installing into a temporary directory
      Running `maturin pep517 build-wheel -i
      C:\a\_temp\setup-uv-cache\builds-v0\.tmpv7xmj3\Scripts\python.exe
      --compatibility off`

      [stderr]
      Python reports platform: win-amd64
      Computed rustc target triple: x86_64-pc-windows-msvc
      Installation directory:
      C:\Users\runneradmin\AppData\Local\puccinialin\puccinialin\Cache
      Rustup already downloaded
      Installing rust to
      C:\Users\runneradmin\AppData\Local\puccinialin\puccinialin\Cache\rustup
      warn: It looks like you have an existing rustup settings file at:
      warn:
      C:\Users\runneradmin\AppData\Local\puccinialin\puccinialin\Cache\rustup\settings.toml
      warn: Rustup will install the default toolchain as specified in the
      settings file,
      warn: instead of the one inferred from the default host triple.
      info: profile set to minimal
      info: default host triple is aarch64-pc-windows-msvc
      warn: Updating existing toolchain, profile choice will be ignored
      info: syncing channel updates for stable-aarch64-pc-windows-msvc
      info: default toolchain set to stable-aarch64-pc-windows-msvc
      Checking if cargo is installed
      🍹 Building a mixed python/rust project
      🔗 Found pyo3 bindings
      🐍 Found CPython 3.10 at
      C:\a\_temp\setup-uv-cache\builds-v0\.tmpv7xmj3\Scripts\python.exe
      📡 Using build options profile, features, locked from pyproject.toml
         Compiling target-lexicon v0.13.5
         Compiling proc-macro2 v1.0.106
         Compiling unicode-ident v1.0.24
         Compiling quote v1.0.45
         Compiling libc v0.2.184
         Compiling rustversion v1.0.22
         Compiling heck v0.5.0
      '+sha2' is not a recognized feature for this target (ignoring feature)
      '+neon' is not a recognized feature for this target (ignoring feature)
      '+crc' is not a recognized feature for this target (ignoring feature)
      error[E0463]: can't find crate for `core`
        |
        = note: the `x86_64-pc-windows-msvc` target may not be installed
        = help: consider downloading the target with `rustup target add
      x86_64-pc-windows-msvc`
        = help: consider building the standard library from source with `cargo
      build -Zbuild-std`

      For more information about this error, try `rustc --explain E0463`.
      error: could not compile `libc` (lib) due to 1 previous error
      warning: build failed, waiting for other jobs to finish...
      💥 maturin failed
        Caused by: Failed to build a native library through cargo
        Caused by: Cargo build finished with "exit code: 101": `"cargo"
      "rustc" "--profile" "release" "--features" "pyo3/extension-module"
      "--target" "x86_64-pc-windows-msvc" "--message-format"
      "json-render-diagnostics" "--locked" "--manifest-path"
      "C:\\a\\_temp\\setup-uv-cache\\sdists-v9\\path\\938a27ddb9b6d1d7\\LWa855XvUxlnvSQj49vzG\\src\\Cargo.toml"
      "--lib" "--crate-type" "cdylib" "--" "-C" "strip=symbols"`
      Error: command ['maturin', 'pep517', 'build-wheel', '-i',
      'C:\\a\\_temp\\setup-uv-cache\\builds-v0\\.tmpv7xmj3\\Scripts\\python.exe',
      '--compatibility', 'off'] returned non-zero exit status 1

      hint: This usually indicates a problem with the package or the build
      environment.
```

From the error message above, two pieces of information stood out.

```
warn: Rustup will install the default toolchain as specified in the
settings file,
warn: instead of the one inferred from the default host triple.
info: profile set to minimal
info: default host triple is aarch64-pc-windows-msvc
warn: Updating existing toolchain, profile choice will be ignored
info: syncing channel updates for stable-aarch64-pc-windows-msvc
info: default toolchain set to stable-aarch64-pc-windows-msvc
```

`rustup` is incorrectly installing the ARM toolchain, even when it has detected that Python is on x64 via SOABI.

```toml
[stderr]
Python reports platform: win-amd64
Computed rustc target triple: x86_64-pc-windows-msvc
```

This tells me that `puccinialin` is not following the architecture of Python.

--- a/src/puccinialin/_setup_rust.py
+++ b/src/puccinialin/_setup_rust.py
@@ -62,7 +62,11 @@ def download_rustup(
 
 
 def install_rust(
-    rustup_init: Path, rustup_home: Path, cargo_home: Path, file: typing.IO
+    rustup_init: Path,
+    rustup_home: Path,
+    cargo_home: Path,
+    file: typing.IO,
+    target_triple: str,
 ):
     """Run rustup to install rust and cargo"""
     print(f"Installing rust to {rustup_home}", file=file)
@@ -73,6 +77,8 @@ def install_rust(
             "--no-modify-path",
             "--profile",
             "minimal",
+            "--default-host",
+            target_triple,
         ],
         env={
             **os.environ,
@@ -128,7 +134,7 @@ def setup_rust(
     # https://github.com/rust-lang/rustup/issues/4607
     lock = FileLock(rustup_init.with_suffix(".lock"))
     with lock:
-        install_rust(rustup_init, rustup_home, cargo_home, file)
+        install_rust(rustup_init, rustup_home, cargo_home, file, target_triple)
 
     # Step 4: Construct and return a dict of changed environment variables to use this rust installation
     new_path = f"{cargo_home.joinpath('bin')}{target.env_path_separator()}{os.environ.get('PATH')}"

